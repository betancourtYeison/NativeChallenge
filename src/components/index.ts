export * from './Body/Body';
export * from './Header/Header';
export * from './Item/Item';
