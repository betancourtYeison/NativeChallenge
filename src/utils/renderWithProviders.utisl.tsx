import React from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react-native';

import { banksSlice } from '../redux/features/banksSlice';

export const renderWithProviders = (
  ui: any,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { banks: banksSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: any = {},
) => {
  const Wrapper = ({ children }: any) => {
    return <Provider store={store}>{children}</Provider>;
  };

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
