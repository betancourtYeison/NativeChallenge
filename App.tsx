/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import { SafeAreaView, StatusBar, useColorScheme } from 'react-native';

import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';

import { Body, Header } from './src/components';
import { store, persistor } from './src/redux/store';
import { theme } from './src/assets/colors/theme';

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';
  const backgroundStyle = { backgroundColor: false ? theme.darker : theme.primary };
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <SafeAreaView style={backgroundStyle}>
          <StatusBar
            barStyle={isDarkMode ? 'light-content' : 'dark-content'}
            backgroundColor={backgroundStyle.backgroundColor}
          />
          <Header />
          <Body />
        </SafeAreaView>
      </PersistGate>
    </Provider>
  );
};

export default App;
