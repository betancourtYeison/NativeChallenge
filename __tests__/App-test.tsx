/**
 * @format
 */

import 'react-native';
import React from 'react';
import '@testing-library/jest-native/extend-expect';

// Note: test renderer must be required after react-native.
import { renderWithProviders } from '../src/utils';
import App from '../App';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderWithProviders(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
