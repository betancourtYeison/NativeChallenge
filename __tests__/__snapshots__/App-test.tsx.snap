// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#00A3E0",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FFFFFF",
        "flexDirection": "column",
        "height": 200,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        {
          "height": 150,
          "width": "100%",
        }
      }
      testID="logo"
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 35,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Bank List
    </Text>
  </View>
  <ActivityIndicator
    color="#00A3E0"
    size="large"
  />
</RCTSafeAreaView>
`;
